src/uwtools/api/file.py

```diff
@@ -24,29 +26,30 @@ def copy(
     """
     Copy files.
 
-    :param target_dir: Path to target directory
     :param config: YAML-file path, or ``dict`` (read ``stdin`` if missing or ``None``).
+    :param target_dir: Path to target directory.
     :param cycle: A datetime object to make available for use in the config.
     :param leadtime: A timedelta object to make available for use in the config.
-    :param keys: YAML keys leading to file dst/src block
-    :param dry_run: Do not copy files
+    :param keys: YAML keys leading to file dst/src block.
+    :param dry_run: Do not copy files.
     :param stdin_ok: OK to read from ``stdin``?
-    :return: ``True`` if no exception is raised
+    :return: ``True`` if all copies were created.
     """
-    _FileCopier(
-        target_dir=Path(target_dir),
+    copier = _FileCopier(
+        target_dir=Path(target_dir) if target_dir else None,
         config=_ensure_data_source(_str2path(config), stdin_ok),
         cycle=cycle,
         leadtime=leadtime,
         keys=keys,
         dry_run=dry_run,
-    ).go()
-    return True
+    )
+    assets: list[Asset] = copier.go()  # type: ignore
+    return all(asset.ready() for asset in assets)
```

Comment:
This is the new `bool` return value for `uwtools.api.file.copy()`: `True` if all the copies were made, `False` otherwise. This allows the return value to be used by higher-level tasks to decide whether or not it is safe to use the expected copies.